components/webhook-mgt/org.wso2.carbon.identity.webhook.management/src/main/java/org/wso2/carbon/identity/webhook/management/internal/service/impl/WebhookManagementServiceImpl.java

```diff
@@ -139,9 +139,22 @@ public Webhook updateWebhook(String webhookId, Webhook webhook, String tenantDom
         if (!existingWebhook.getEndpoint().equals(webhook.getEndpoint()) &&
                 daoFACADE.isWebhookEndpointExists(webhook.getEndpoint(), tenantId)) {
             throw WebhookManagementExceptionHandler.handleClientException(
-                    ErrorMessage.ERROR_CODE_WEBHOOK_ALREADY_EXISTS);
+                    ErrorMessage.ERROR_CODE_WEBHOOK_ENDPOINT_ALREADY_EXISTS, webhook.getEndpoint());
         }
-        daoFACADE.updateWebhook(webhook, tenantId);
+        Webhook updatedWebhook = new Webhook.Builder()
+                .uuid(webhookId)
+                .endpoint(webhook.getEndpoint())
+                .name(webhook.getName())
+                .secret(webhook.getSecret() != null ? webhook.getSecret() : existingWebhook.getSecret())
```

Comment:
This logic doesn't work as per the way you have implemented when retrieving the webhook secret is not returned even from DAO